ddtrace/debugging/_probe/model.py

```diff
@@ -246,7 +247,59 @@ class SpanFunctionProbe(Probe, FunctionLocationMixin, SpanProbeMixin, ProbeCondi
 FunctionProbe = Union[LogFunctionProbe, MetricFunctionProbe, SpanFunctionProbe]
 
 
+class ProbeFactory(object):
+    __line_class__ = None  # type: Optional[Type[LineProbe]]
+    __function_class__ = None  # type: Optional[Type[FunctionProbe]]
+
+    @classmethod
+    def build(cls, args, where, evaluate_at=None):
+        # type: (Dict[str, Any], Dict[str, Any], Optional[ProbeEvaluateTimingForMethod]) -> Any
+        if where.get("sourceFile", None) is not None:
+            if cls.__line_class__ is None:
+                raise TypeError("Line probe type is not supported")
+
+            args["source_file"] = where["sourceFile"]
```

Comment:
Few thoughts about this refactor.
1. I like the Factory pattern - great step forward
2. this logic "sourceFile" and "where" are specific for remote-config.
3. the specific logic for each probe lives under remote-config file. 
4. both ProbeType and remoteconfig doing switch on probe-type - looks like something we can improve

So, my take on this:
1. Keep the factories under remoteconfig file.
2. each specific factory LogProbeFatory should also handle it own specific arguments.

that way the code at remoteconfig file would be:
```
factory = ProbeType.get_factory(object)
probe = factory.createProbe(object)
```